CMakeLists.txt

```diff
@@ -200,6 +204,42 @@ if(CMAKE_SYSTEM_NAME MATCHES "Linux")
         endif()
         SET(CMAKE_C_LINK_EXECUTABLE "<CMAKE_LINKER> <LINK_FLAGS> <OBJECTS> -o <TARGET> -Wl,--start-group <LINK_LIBRARIES> -Wl,--end-group")
 
+    elseif(TOOLCHAIN STREQUAL "ARM_CLANG")
+        ADD_DEFINITIONS(-DLIBSPDM_ARMCLANG_BUILD=1)
```

Comment:
I dont think we allow this LIBSPDM_ARMCLANG_BUILD. Too wried.

Can you just use "__clang__" for CLANG compiler, and "LIBSPDM_CPU_xxx" for different ARCH?